lib/puppet/application/apply.rb

```diff
@@ -185,17 +185,25 @@ def main
       facts.name = Puppet[:node_name_value]
     end
 
-    configured_environment = Puppet.lookup(:current_environment)
+    # Find our Node
+    unless node = Puppet::Node.indirection.find(Puppet[:node_name_value])
+      raise "Could not find node #{Puppet[:node_name_value]}"
```

Comment:
Question is valid, though I would prefer to postpone solution to an indipendent *maint* style branch. This is old code and appears in the patch only because of a whitespace change. This branch should not touch it.